src/core/lib/event_engine/thread_pool/work_stealing_thread_pool.cc

```diff
@@ -271,6 +310,37 @@ void WorkStealingThreadPool::WorkStealingThreadPoolImpl::Postfork() {
   Start();
 }
 
+void WorkStealingThreadPool::WorkStealingThreadPoolImpl::TrackThread(
+    gpr_thd_id tid) {
+  grpc_core::MutexLock lock(&thd_set_mu_);
+  thds_.insert(tid);
+}
+
+void WorkStealingThreadPool::WorkStealingThreadPoolImpl::UntrackThread(
+    gpr_thd_id tid) {
+  grpc_core::MutexLock lock(&thd_set_mu_);
+  thds_.erase(tid);
+}
+
+void WorkStealingThreadPool::WorkStealingThreadPoolImpl::DumpStacksAndCrash() {
+  grpc_core::MutexLock lock(&thd_set_mu_);
+  gpr_log(GPR_ERROR,
+          "Pool did not quiesce in time, gRPC will not shut down cleanly. "
+          "Dumping all %zu thread stacks.",
+          thds_.size());
+  for (const auto tid : thds_) {
+    grpc_core::Thread::Signal(tid, kDumpStackSignal);
```

Comment:
This is likely to have some unintended consequences when we're linked into someone elses binary -- signals are a limited resource, and their assignment needs to be at the whim of the binary owner, not some random library they link in.

Consequently I'd recommend placing this under some kind of environment variable and only enabling it when we're running under our own CI.

Reply:
Done. I've added an internal env var that controls this behavior, and all `grpc_cc_test`s set it.

I also updated the logic to check the timeout relative to the last count change. Some stress tests may take a while to quiesce, which is fine. Generally in these circumstances, the pool will grow & shrink, so I'm using the count to indicate wedgedness.